roles/servicetelemetry/tasks/component_qdr.yml

```diff
@@ -1,4 +1,147 @@
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-openstack-credentials"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect"
+          dnsNames:
+            - "{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local"
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-ca"
+          secretName: "{{ meta.name }}-interconnect-openstack-credentials"
+
+  - name: Create inter-router CA issuer
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Issuer
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-ca"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          ca:
+            secretName: "{{ meta.name }}-interconnect-inter-router-ca"
+
+  - name: Create inter-router CA certificate
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-ca"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect-inter-router-ca"
+          isCA: true
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-selfsigned"
+          secretName: "{{ meta.name }}-interconnect-inter-router-ca"
+
+  - name: Create inter-router credentials certificate
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-credentials"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect"
+          dnsNames:
+            - "{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local"
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-inter-router-ca"
+          secretName: "{{ meta.name }}-interconnect-inter-router-credentials"
+  when: interconnect_manifest is not defined
+
+- name: Create Interconnect SASL ConfigMap
+  k8s:
+    state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+    definition:
+      kind: ConfigMap
+      apiVersion: v1
+      metadata:
+        name: "{{ meta.name }}-interconnect-sasl-config"
+        namespace: "{{ meta.namespace }}"
+        labels:
+          application: "{{ meta.name }}-interconnect"
+          interconnect_cr: "{{ meta.name }}-interconnect"
+      data:
+        qdrouterd.conf: |
+          pwcheck_method: auxprop
+          auxprop_plugin: sasldb
+          sasldb_path: /tmp/qdrouterd.sasldb
+  when: interconnect_manifest is not defined
```

Comment:
I guess this is part of the `generate_...` parameters normally provided to `Interconnect` manifests so we need to generate this now.

Reply:
> I can't comment on L53 / 196 below, but I'm thinking we need to remove those generate* properties?

Oops! Did I not disable them? Definitely need to do that (although the Interconnect Operator knows that the `certmanager.k8s.io` API doesn't exist, and will say that it has the inability to create certificates, so it seems to handle the Operator / CRD missing gracefully, but we should definitely be explicit in our `generate...` configuration. Nice catch, will fix that tomorrow.

For the SASL stuff, I believe it comes from the `interRouterListeners` configuration, which in `component_qdr.yml` on L186 has `saslMechanisms: EXTERNAL`. As a result the Interconnect workload would fail to launch due to missing ConfigMap, so I just made sure it was available as it was previously in STF 1.4.

I haven't looked at the code, but my theory is that since the AMQ Cert Manager (certmanager.k8s.io) isn't available, the code path that would have generated this ConfigMap in Interconnect Operator is not run, and results in the workload unable to load a volume it is expecting. Adding the ConfigMap here results in the router workloads coming up as intended.

Reply:
I have changed the `generate_...` commands to be `false`.